test/gtest/uct/test_p2p_rma.cc

```diff
@@ -62,18 +62,25 @@ ucs_status_t uct_p2p_rma_test::get_zcopy(uct_ep_h ep, const mapped_buffer &sendb
 void uct_p2p_rma_test::test_xfer(send_func_t send, size_t length,
                                  direction_t direction)
 {
-    mapped_buffer sendbuf(length, SEED1, sender(), 1);
-    mapped_buffer recvbuf(length, SEED2, receiver(), 3);
-
-    blocking_send(send, sender_ep(), sendbuf, recvbuf, true);
-    if (direction == DIRECTION_SEND_TO_RECV) {
-        sendbuf.pattern_fill(SEED3);
-        wait_for_remote();
-        recvbuf.pattern_check(SEED1);
-    } else if (direction == DIRECTION_RECV_TO_SEND) {
-        recvbuf.pattern_fill(SEED3);
-        sendbuf.pattern_check(SEED2);
-        wait_for_remote();
+    for (int mem_type = 0; mem_type < UCT_MD_MEM_TYPE_LAST; mem_type++) {
```

Comment:
can we make this iteration inside uct_p2p_test::test_xfer_multi instead?
e.g
1. rename test_xfer_multi() to test_xfer_multi_mem_type(..., mem_type)
2. test_xfer_multi() will iterate over memtype and call test_xfer_multi_mem_type()
3. test_xfer_multi() will print to the output ("ms") the tested memory type, in the beginning of the line, so we could see which memtypes are being tested

Reply:
I wanted to restrict this to rma, if I move iteration to uct_p2p_test::test_xfer_multi and pass mem_type to  subclasse (rma,am ..etc) test_xfer, then I have to ignore mem_type in uct_p2p_am_test.  If I do that same test will run twice with host memory because of mem type iteration.
For example  IB TL support cuda memory but not support short, bcopy..
is there anyway to handle this scenario?

Reply:
pushed a solution. please review.

Reply:
but we do want to test IB's `am_zcopy` with cuda memory, right?